### ISSUE TITLE
Regarding: Enyaq / Skoda-Connect

### ISSUE THREAD

**Issue by MarcelMertens** [skodaconnect/skodaconnect#126]
Hi Everyone,

i got a brand new Enyaq (Germany). Created a https://identity.vwgroup.io/ account, used the MySkoda (**not** Essential) App to connect to my Enyaq, Everything works fine.
But in skoda-connect.com no car is shown. Also the HA HACS PlugIn reports that no car is avaiable.
When adding a car on skoda-connect.com i get the info that i have to use the "MySkoda" app.
As i already configured thed MySkoda app i assume that i have to use the "old" MySkoda Essential App. But after i entered my VIN i got an connection error...
Is the "Essential" app correct to add my Enyaq to the skoda-connect.com?

**Comment by dvx76** [comment]
Hi @MarcelMertens ,

Nobody is really quite sure what the _correct_ way is. Sometimes I wonder if even VW/Skoda does.

For the HA integration to work what we know is that you need to connect using the old (Essentials) app at least once so that you are prompted to accept the terms and conditions. The integration uses the "old" API (same as the Essentials app) and this T&C acceptance step isn't implemented (and without it the token exchange sequence fails).

The new MySkoda app uses a (at least partially different) API which hasn't been reversed and implemented in HA yet. We suspect that at some point the old (Essentials) app will stop working, and potentially with it the HA integration.

It _could_ be that something like this is now rolled out for new accounts/vehicles where the old Essentials app just isn't allowed anymore. But you'd be the first case we know about.

Unfortunately there isn't much we can do to help. You need to first get the Essentials app working. Are you able to share the exact (screenshot or copy/paste) connection error you get in the Essentials app?

**Comment by MarcelMertens** [comment]
<img width="1574" alt="image" src="https://github.com/skodaconnect/skodaconnect/assets/20799825/7d3df091-5e9a-4d19-a721-870ce9966b74">
Activation Process from Left to Right..
If you need a translation, let me know.
If it would help i can do some API exploration with Burp Suite from the new "MySkoda" App
I have some ReST API / backend stack knowledge, so i can help if you want...

**Comment by dvx76** [comment]
Thanks that's a great level of detail.

I just logged out and back in with the Essentials app so whatever is your problem it's specific to the initial registration sequence. It might be worth grabbing the actual request and response which leads to that last screenshot (is it really a connection failure or is it a negative HTTP response 🤔 ). Other than that the only thing I can suggest is to contact Skoda (support@skoda-auto.com). They will tell you to just use the new MySkoda app. You could try to say you have an old phone and only the old Essentials app works on it ... but it's a long shot. Or maybe it actually _is_ a temporary server-side issue and it goes away after a few days. Wouldn't be the first time...

Wrt inspecting the new MySkoda app, first of all many thanks for proposing your help. I have some network traces lying around already but it looks like they switched from a REST API to Websockets and unfortunately none of the current maintainers have much experience with this ☹️ .

There's a Discord linked in the README. Feel free to join and I can add you to the dev channel.

**Comment by mellowtronik** [comment]
@MarcelMertens 
Hello, 
I have exactly the same problem 😕.  I'm from France.  I called my Skoda sales person and he told me it was a general problem and they didn't know if it would be fixed 🥹.  It would be nice if a developer did something with the Myskoda app because the old version of Skoda Connect is going to disappear.  😕

**Comment by MatzeLoCal** [comment]
Hi, 
I also have the same problem. The "new" skoda.app works without any problem, but the older mySkoda has the same issues as shown above. I have a MY2024 Enyaq 85.

**Comment by ChrisZ32** [comment]
Not sure it makes any difference but just to clarify, I've got this problem and my car is MY2021 so it seems unrelated to the age of the car, seems to be around the registration flow of the old app. So cars already registered work, but you cannot register enyaqs of any age on the old app now

**Comment by JannickWe** [comment]
I have the same problem with a new Enyaq 60 MY24. 

I went through the config flow of the MySkoda Essentials app (Android & iOS) which always results in the "connection failed" error, too. However, when I finished charging yesterday, both the new (working) MySkoda and the MySkoda essentials App sent me a push notification at the same time "Charging is completed". There is no car listed in the Essentials app, but it seems to be linked anyway. 

Additionally, i enabled looking for the Home Assistant integration. My interpretation is that the consent is given but no car is found.

```
2024-05-09 17:14:43.005 INFO (MainThread) [skodaconnect.connection] Initiating new login
2024-05-09 17:14:44.511 INFO (MainThread) [skodaconnect.connection] Token for technical verified OK.
2024-05-09 17:14:45.163 INFO (MainThread) [skodaconnect.connection] Token for connect verified OK.
2024-05-09 17:14:46.070 ERROR (MainThread) [custom_components.skodaconnect.config_flow] Fetch vehicles failed with error: Invalid token "None"

```

**Comment by MatzeLoCal** [comment]
Followed the advise by @dvx76  and contacted the skoda support. The responded very quickly but with a strange answer.
They told me that I was not using the latest Version of the App. The Version I use is 5.8.2 and they told me to use 5.8.4 … but there is no 5.8.4 on the Apple AppStore :/

**Comment by mellowtronik** [comment]
> J'ai suivi les conseils de@dvx76 et j'ai contacté le support Skoda. Ils ont répondu très rapidement mais avec une réponse étrange. Ils m'ont dit que je n'utilisais pas la dernière version de l'application. La version que j'utilise est la 5.8.2 et ils m'ont dit d'utiliser la 5.8.4… mais il n'y a pas de 5.8.4 sur l'AppStore d'Apple :/

Bonjour, j'utilise la version 5.8.5 de l'ancienne version et ça ne fonctionne toujours pas.
Mon commercial Skoda m'a dit que c'était un problème international et ils ne savent pas si ça va être réparé 😭
![Screenshot_2024-05-10-14-01-20-760_com android vending](https://github.com/skodaconnect/skodaconnect/assets/91826922/9a27b3c2-6e87-4709-b53f-3ff550aa9c51)

**Comment by Mheel80** [comment]
> > J'ai suivi les conseils de@dvx76 et j'ai contacté le support Skoda. Ils ont répondu très rapidement mais avec une réponse étrange. Ils m'ont dit que je n'utilisais pas la dernière version de l'application. La version que j'utilise est la 5.8.2 et ils m'ont dit d'utiliser la 5.8.4… mais il n'y a pas de 5.8.4 sur l'AppStore d'Apple :/
> 
> Bonjour, j'utilise la version 5.8.5 de l'ancienne version et ça ne fonctionne toujours pas. Mon commercial Skoda m'a dit que c'était un problème international et ils ne savent pas si ça va être réparé 😭 ![Screenshot_2024-05-10-14-01-20-760_com android vending](https://private-user-images.githubusercontent.com/91826922/329571556-9a27b3c2-6e87-4709-b53f-3ff550aa9c51.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU3MTA3MTgsIm5iZiI6MTcxNTcxMDQxOCwicGF0aCI6Ii85MTgyNjkyMi8zMjk1NzE1NTYtOWEyN2IzYzItNmU4Ny00NzA5LWI1M2YtM2ZmNTUwYWE5YzUxLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE0VDE4MTMzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0OWQxOThmZWFjNDFjZmRlYjlhZjlmM2VkYzc5MDgyMzBkZGJlNzNjM2RjYTM1YmI0YmVkZGE1MDZhZDVhOWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5zBk7LXVO0v7-Gmm2eaUsSTe_q85Vda500wpm38iPXk)

But that is for android not IOS. In IOS it still 5.8.2. I have the same issue. I have tryed it on a android But it doesn't help. Same issue there, No car.

**Comment by matlimatli** [comment]
I'm not sure if this might give some additional insight or not.

I'm just now setting up Home Assistant on a new host. I am not able to add my Skoda to this instance (getting `2024-05-15 22:55:52.122 ERROR (MainThread) [custom_components.skodaconnect.config_flow] Login failed with error: The terms and conditions must be accepted first at "https://www.skoda-connect.com/"`, but at the same time everything still works fine on the old host. 

In other words, I have successfully accepted the T&C previously, which seems to be enough for the actual data updates, but not for the authentication of a new instance.

**Comment by matlimatli** [comment]
I can also add that by accepting the updated terms and conditions in the 5.8.5 app (on Android, does not appear to be available on iOS yet), I was able to add my Skoda to the new HA instance.

**Comment by Mheel80** [comment]
> I can also add that by accepting the updated terms and conditions in the 5.8.5 app (on Android, does not appear to be available on iOS yet), I was able to add my Skoda to the new HA instance.

Is the car visibel in essential app?
I just tried again with an android. It didn’t work.

**Comment by matlimatli** [comment]
> Is the car visibel in essential app? I just tried again with an android. It didn’t work.

Yes, visible in both essential and regular app. On both Android and iOS. However, there were some extra privacy nag screens that I had to click through in the car this morning. I'm unsure if they were related.

**Comment by Steve-Lammas** [comment]
Similar issue with 2023 Skoda Enyaq 80 Sportline Plus acquired on Sunday. 

**New** Skoda connect account set up on Android 5.8.5 essentials app with no previous acceptance of T&Cs.

Process stalls at "Checking the status of the vehicle activation" 

Revoking T&Cs on portal and reinstalling the app doesn't resolve the issue.

Alternative app does work, albeit with reduced feature set.

Response to my disgruntled e-mail from Skoda Support for Digital Services - "We kindly inform you that MyŠKODA Essentials app is not compatible with Enyaq model. We recommend installing MyŠKODA app as this app is developed for Enyaq model specifically."

**Comment by mellowtronik** [comment]
Bonsoir,
Mail de Skoda. Skoda Connect va casser de fonctionner d'ici juillet 2024 😭
![Screenshot_2024-05-28-22-02-34-916_com microsoft office outlook](https://github.com/skodaconnect/skodaconnect/assets/91826922/e15a5dfd-1fca-4628-94c7-957fc900a2e2)

**Comment by dvx76** [comment]
Hi,

Most users have now received the same email wrt the Essentials app EOL. We also know that for new vehicles/accounts the Essentials app, and consequently this integration, is already not working.

There are only a few active developers working on this integration but we are aware of this and we are (slowly) working on making the integration compatible with the new MySkoda app/API. There are no timelines and in fact no guarantee we will get this working. Please be patient and do not open new issues reporting the same thing.
